### PR TITLE
Fixed imports/exports that are illegal identifiers in the es output

### PR DESCRIPTION
**Pull request by Andarist**
This PR contains:

- [x] bugfix

Are tests included?

- [x] yes

Breaking Changes?

- [x] no

### Description

This is legal JS and it should produce legal ES output 😉 
```ts
const foo = 1
export { foo as "🔥" }
```

### CODE REVIEW

**Review by lukastaegert — COMMENTED** [review]
Thanks for tackling this! While the fix is fine, there are some more import/reexport cases I think need to be covered, would you be willing to address them as well?

File: test/chunking-form/samples/export-illegal-identifier/_config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (Andarist) [reply test/chunking-form/samples/export-illegal-identifier/_config.js]: moved this to the requested location

**Comment by Andarist** [comment]
> Thanks for tackling this! While the fix is fine, there are some more import/reexport cases I think need to be covered, would you be willing to address them as well?

I've fixed the mentioned cases and a couple more. I **think** that I'm wrapping almost every location that has to be wrapped with the `safeExportName` in the `es` finalizer. There are locations in there that might look like then need wrapping at the first glance but I think that they refer to local bindings and those always have to be valid so I left them untouched (I'm only trying to wrap what has to be wrapped and I try to avoid doing redundant "work" where it's not needed).

However, I intentionally left one spot not wrapped with this. I discovered that `acorn`+`acorn-import-assertions` fails to parse this:
```ts
export * as '🙈' from 'external';
```

So I couldn't add this to a test right now. I could wrap the required call site preemptively but I would prefer to only do it once `acorn-import-assertions` fixes [the issue](https://github.com/xtuc/acorn-import-assertions/issues/20).

Also, thanks to the fact that you have used math operators in your proposed test cases... I discovered that this doesn't work properly either (and I don't plan to work on fixing it rn 😉):
```ts
// input
import { '*' as test } from 'external'
console.log(test)

// actual
import * as banan from 'external';
console.log(test)

// expected
import { '*' as _safe } from 'external'
console.log(_safe)
```

**Review by lukastaegert — APPROVED** [review]
Looks great! One might consider using some sanitizer function instead of `_safe`, but that is not really important as the deconflicting logic works as intended. I will merge this one now.

**Comment by JochenDiekenbrock** [comment]
I believe this change breaks my builds. The run fine with 3.20.4 and break with 3.20.5.

I don't know what to look for and I can't provide a code to reproduce the problem, since it's private. Maybe you have an idea anyway or can direct me how to diagnose to root cause?

I'm getting 
[!] Error: Cannot overwrite a zero-length range – use appendLeft or prependRight instead
    at MagicString.update (/home/jochen/WebstormProjects/shared/node_modules/rollup/dist/shared/rollup.js:2142:10)
    
I added some logging and found that MagicString.update is invoked with "_safe" and
that getSafeName finds a lot of invalid identifiers. The last call before the error above is called with
baseName 2038 and the generated safeName is "_safe"

**Comment by Andarist** [comment]
Might be super hard to find this if you don't provide a repro case. Perhaps try to slap an extra global counter on each `_safe` and then check which one ends up crashing `MagicString` there. This could help you to identify which exact invalid identifier relates to this crash - this could help you to minimize the repro further.

**Comment by lukastaegert** [comment]
I think I will roll back the feature until we figured out what is wrong. Nicely we know have a reproduction at #4945

**Comment by webcarrot** [comment]
@Andarist 

I'm using `rollup-plugin-dts` and I started to have the same issue.

# Minimal example

## rollup.config.mjs

```js
import dts from 'rollup-plugin-dts';
export default [
	{
		input: './src/index.ts',
		output: [{ file: `dist/index.d.ts`, format: 'es' }],
		plugins: [dts()],
	},
];
```

## src/index.ts

```js
import { Dispatcher } from 'undici';

export function makeDispatcherProvider() {
	let d: Dispatcher;
	return {
		getDispatcher() {
			return d;
		},
		setDispatcher(newDispatcher: Dispatcher) {
			d = newDispatcher;
		},
	};
}
```

## `rollup -c ./rollup.config.mjs`

```
./src/index.ts → dist/index.d.ts...
[!] Error: Cannot overwrite a zero-length range – use appendLeft or prependRight instead
    at MagicString.update (/DIR/node_modules/rollup/dist/shared/rollup.js:2142:10)
    at MagicString.overwrite (/DIR/node_modules/rollup/dist/shared/rollup.js:2131:15)
    at Identifier.render (/DIR/node_modules/rollup/dist/shared/rollup.js:8762:22)
    at AssignmentPattern.render (/DIR/node_modules/rollup/dist/shared/rollup.js:9439:19)
    at FunctionDeclaration.render (/DIR/node_modules/rollup/dist/shared/rollup.js:6346:28)
    at renderStatementList (/DIR/node_modules/rollup/dist/shared/rollup.js:8913:35)
    at Program.render (/DIR/node_modules/rollup/dist/shared/rollup.js:12199:13)
    at Module.render (/DIR/node_modules/rollup/dist/shared/rollup.js:14105:18)
    at Chunk.renderModules (/DIR/node_modules/rollup/dist/shared/rollup.js:16554:41)
    at Chunk.render (/DIR/node_modules/rollup/dist/shared/rollup.js:16062:111)
```

**Pull request merged**